docs/4.0/aws_oss_guide.md

```diff
@@ -0,0 +1,342 @@
+3. Once a machine has been bootstrapped with SSH Keys, there isn't an easy way to 
+  add new keys and delegate access. 
+
+**Which Services can I use Teleport with?**
+
+You can use Teleport for all the services that you would SSH into. This guide is focused
+on EC2. We have a short blog post on using Teleport with [EKS](https://gravitational.com/blog/teleport-aws-eks/). We plan to expand the guide based on feedback but will plan to add instructions
+for the below. 
+
+- RDS
+- Detailed EKS
+- Lightsail
+- Fargate
+- AWS ECS 
+
+## Teleport Introduction
+
+This guide will cover how to setup, configure and run Teleport on [AWS](https://aws.amazon.com/). 
+ 
+#### AWS Services required to run Teleport in HA  
+
+- [EC2 / Autoscale](#ec2-autoscale)
+- [DynamoDB](#dynamodb)
+- [S3](#s3)
+- [Route53](#route53) 
+- [NLB](#nlb-network-load-balancer)
+- [IAM](#iam)
+- [ACM](#acm)
+- [SSM](#aws-systems-manager-parameter-store) 
+
+We recommend setting up Teleport in high availability mode (HA). In HA mode DynamoDB 
+stores the state of the system and S3 will store audit logs. 
+
+![AWS Intro Image](img/aws/aws-intro.png)
+
+### EC2 / Autoscale
+To run Teleport in a HA configuration we recommend using m4.large instances and it's best
+practice to separate the proxy and authentication server, creating auto-scale groups 
+for both machines. We've a range of AMIs that have Teleport already built in it. 
+Instructions on using these [AMIs are below](#).
+
+### DynamoDB
+DynamoDB is a  key-value and document database that delivers single-digit millisecond 
+performance at any scale. For large clusters you can provision usage but for smaller 
+deployments you can leverage DynamoDBs auto-scale. 
+
+Teleport 4.0 leverages [DynamoDB streaming feature](
+https://github.com/gravitational/teleport/issues/2430). When turning this on, you'll need
+to specify `New Image` from the streaming options. DynamoDB back-end supports two 
+types of Teleport data:
+
+* Cluster state
+* Audit log events
+
+
+### S3 
+Amazon Simple Storage Service (Amazon S3) is an object storage service that offers
+industry-leading scalability, data availability, security, and performance. In 
+this setup of Teleport S3 will be the store of recorded sessions. 
+ 
+We recommend using Amazon S3 Standard. 
+
+!!! tip "Tip":
+    S3 provides [Amazon S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lock.html),
+    which is useful for customers deploying Teleport in regulated environments. 
+
+### Route53
+Route53 is a highly available Domain Name System (DNS) provided by AWS. It'll be 
+needed to setup a URL for the proxy, we recommend using a sub-domain. 
+
+e.g. `teleport.acmeinc.com`
+
+### NLB: Network Load Balancer
+AWS provides many different load balancers. To setup Teleport, we recommend 
+using a Network Load Balancer.  Network Load Balancers provides TLS for the Teleport 
```

Comment:
NLBs don't actually provide TLS themselves, they just proxy a TCP connection through and Teleport itself will still handle the TLS.

Reply:
I actually expanded this to outline how we use ALB and NLB as load balancers.